src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -89,81 +124,453 @@ static DamageTable D_808B10C8 = {
+            Enemy_StartFinishingBlow(globalCtx, &this->actor);
+        }
+        if (this->actor.colChkInfo.damageEffect == 0xD) {
+            return true;
+        }
+        if (this->actor.colChkInfo.health != 0) {
+            this->enemyCollider.base.atFlags &= ~1;
+        }
+        this->enemyCollider.base.atFlags &= ~2;
+        if (this->actor.colChkInfo.damageEffect == 4) {
+            this->unk_248 = 0.7f;
+            this->unk_244 = 4.0f;
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0xA2, this->enemyCollider.info.bumper.hitPos.x,
+                        this->enemyCollider.info.bumper.hitPos.y, this->enemyCollider.info.bumper.hitPos.z, 0, 0, 0, 4);
+        }
+        func_808B06D0(this, globalCtx);
+        return true;
+    }
+    return false;
+}
+
+void EnAm_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnAm* this = THIS;
+    s32 pad;
+
+    if (func_808B0B9C(this, globalCtx) == 0) {
+        if ((this->enemyCollider.base.atFlags & 4) != 0) {
+            this->enemyCollider.base.atFlags &= 0xFFF9;
+            if (this->actor.colChkInfo.health == 0) {
+                this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+            } else {
+                func_808B0AD0(this, globalCtx);
+            }
+        }
+    }
+    if (this->returnHomeTimer != 0) {
+        this->returnHomeTimer--;
+    }
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 30.0f, 100.0f, 0x1D);
+    Actor_SetHeight(&this->actor, 64.0f);
+    Collider_UpdateCylinder(&this->actor, &this->enemyCollider);
+    Collider_UpdateCylinder(&this->actor, &this->interactCollider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->enemyCollider.base);
+    if (this->enemyCollider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->enemyCollider.base);
+    }
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->interactCollider.base);
+    if (this->enemyCollider.base.atFlags & AC_ON) {
+        this->actor.flags |= 0x1000000;
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->enemyCollider.base);
+    }
+    Math_StepToF(&this->unk_244, 0.0f, 0.05f);
+    this->unk_248 = (this->unk_244 + 1.0f) * 0.35f;
+    this->unk_248 = (this->unk_248 > 0.7f) ? 0.7f : this->unk_248;
+}
+
+void EnAm_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor) {
+    s32 i;
+    s32 phi_s3;
+    Vec3f* phi_s1;
+    Vec3f* phi_s2;
+    EnAm* this = (EnAm*)actor;
+
+    phi_s2 = 0;
+    phi_s1 = 0;
+    if (limbIndex == 4) {
+        phi_s2 = &this->unk_24C[0];
+        phi_s1 = D_808B1128;
+        phi_s3 = 5;
+    } else if (limbIndex == 0xD) {
+        phi_s2 = &this->unk_24C[9];
+        phi_s1 = D_808B117C;
+        phi_s3 = 4;
+    } else if ((limbIndex == 7) || (limbIndex == 0xA)) {
```

Comment:
```suggestion
    } else if (limbIndex == 14) {
        phi_s2 = &this->unk_24C[9];
        phi_s1 = D_808B117C;
        phi_s3 = 4;
    } else if ((limbIndex == 7) || (limbIndex == 10)) {
```
limbs can be in decimal